boards/cats_rev1Pro/Core/Src/tasks/task_baro_read.c

```diff
@@ -30,33 +32,38 @@ static void read_baro();
 void task_baro_read(void *argument) {
   /* For periodic update */
   uint32_t tick_count, tick_update;
+  uint32_t stage = READ_BARO_TEMPERATURE;
   /* actual measurements from sensor */
   int32_t temperature[3];
   int32_t pressure[3];
 
   tick_count = osKernelGetTickCount();
-  tick_update = osKernelGetTickFreq() / CONTROL_SAMPLING_FREQ;
-
+  tick_update = osKernelGetTickFreq() / (2*CONTROL_SAMPLING_FREQ);
+  prepare_temp();
+  osDelay(5);
   while (1) {
     tick_count += tick_update;
     // Phase 1, get the temperature
-    prepare_temp();
-    osDelay(2);
-    read_baro();
-    osDelay(2);
-    prepare_pres();
-    osDelay(2);
-    read_baro();
-    osDelay(2);
-
-    get_temp_pres(temperature, pressure);
 
-    for (int i = 0; i < 3; i++) {
-      global_baro[i].pressure = pressure[i];
-      global_baro[i].temperature = temperature[i];
-      global_baro[i].ts = tick_count;
+    // Readout the register
+    read_baro();
 
-      record(BARO0 << i, &(global_baro[i]));
+    // Prepare new readout
+    if(stage == READ_BARO_TEMPERATURE){
```

Comment:
also this line doesn't look formatted

Reply:
Could it be that my clang format is not working anymore?